base/applications/rapps/geninst.cpp

```diff
@@ -52,7 +52,10 @@ BOOL IsZipFile(PCWSTR Path)
     zlib_filefunc64_def zff;
     fill_win32_filefunc64W(&zff);
     unzFile hzf = unzOpen2_64(Path, &zff);
-    return hzf && unzClose(hzf) == UNZ_OK;
+    if (!hzf)
+        return FALSE;
+    unzClose(hzf);
+    return TRUE;
```

Comment:
```suggestion
    if (hzf)
        unzClose(hzf);
    return !!hzf;
```
(`!!hzf` or `hzf != NULL` ...)

Reply:
That is certainly not clearer.

Will it generate better code than

```
xor eax, eax
inc eax
```
?

Reply:
I hope the compiler will be clever enough to generate the best code in any case.